### ISSUE TITLE
[Flaky test] [pull-kubernetes-conformance-kind-ga-only-parallel] [sig-node] Events should be sent by kubelets and the scheduler about pods scheduling and running [Conformance]

### ISSUE THREAD

**Issue by ehashman** [kubernetes/kubernetes#106506]
Appears related to https://github.com/kubernetes/kubernetes/issues/94414

I don't see this flaking in the [periodic](https://testgrid.k8s.io/sig-testing-kind#conformance-ga-only) but it appears to be quite flaky in presubmit CI, where this is the only test on the job that fails.

/sig testing node
/kind flake

##### Error text:
```
/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:185
Nov 13 08:28:15.515: Unexpected error:
    <*errors.errorString | 0xc00024e250>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred
/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:243
```
#### Recent failures:
[11/17/2021, 4:25:05 AM pr:pull-kubernetes-conformance-kind-ipv6-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106490/pull-kubernetes-conformance-kind-ipv6-parallel/1460946632680935424)
[11/17/2021, 3:23:18 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460931072895225856)
[11/17/2021, 1:58:59 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460909846491566080)
[11/17/2021, 12:59:17 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460894824499515392)
[11/17/2021, 12:02:17 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460880476934770688)

/kind failing-test

**Comment by aojea** [comment]
I was debugging this test this morning

the apiserver logs shows the kubelet events come too late, the test times out, is like they are queued in the kubelet

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460909846491566080

```
2021-11-17T10:23:05.998659524Z stderr F I1117 10:23:05.998362       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-2864/events" latency="10.620155ms" userAgent="kube-scheduler/v1.23.0 (linux/amd64) kubernetes/96d78e5/scheduler" audit-ID="53fecaf8-c5ca-4c65-a79e-db3477fd8f4d" srcIP="172.18.0.4:40632" resp=201
2021-11-17T10:23:06.932538553Z stderr F I1117 10:23:06.932405       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-2864/serviceaccounts/default/token" latency="28.220359ms" userAgent="Go-http-client/2.0" audit-ID="0d52bbb7-8bb5-4f4b-bb27-e89baf46d2bc" srcIP="172.18.0.2:58228" resp=201
2021-11-17T10:24:27.213880196Z stderr F I1117 10:24:27.213753       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-2864/configmaps" latency="617.475µs" userAgent="kube-controller-manager/v1.23.0 (linux/amd64) kubernetes/96d78e5/system:serviceaccount:kube-system:root-ca-cert-publisher" audit-ID="55c1363d-c474-4790-9337-b6876c2aa9dc" srcIP="172.18.0.4:40642" resp=403
2021-11-17T10:24:51.914796622Z stderr F I1117 10:24:51.914658       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-2864/events" latency="801.816µs" userAgent="kubelet/v1.23.0 (linux/amd64) kubernetes/96d78e5" audit-ID="7f6d9f53-070b-4a23-8907-b007abc0d3a1" srcIP="172.18.0.2:58228" resp=403
2021-11-17T10:24:52.11568577Z stderr F I1117 10:24:52.115491       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-2864/events" latency="1.023306ms" userAgent="kubelet/v1.23.0 (linux/amd64) kubernetes/96d78e5" audit-ID="17e86dd3-27af-4ff8-a3d3-72f067c7bb12" srcIP="172.18.0.2:58228" resp=403
2021-11-17T10:24:52.315100794Z stderr F I1117 10:24:52.314915       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-2864/events" latency="947.248µs" userAgent="kubelet/v1.23.0 (linux/amd64) kubernetes/96d78e5" audit-ID="7fd59fb5-ad7d-4b56-b153-05a9d3030f27" srcIP="172.18.0.2:58228" resp=403
```

but kubelet times doesn't add up
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460909846491566080/artifacts/logs/kind-worker2/journal.log

```
ov 17 10:23:07 kind-worker2 kubelet[250]: I1117 10:23:07.443572     250 event.go:294] "Event occurred" object="events-2864/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288" kind="Pod" apiVersion="v1" type="Normal" reason="Pulled" message="Container image \"k8s.gcr.io/e2e-test-images/agnhost:2.33\" already present on machine"
Nov 17 10:23:07 kind-worker2 kubelet[250]: I1117 10:23:07.614353     250 event.go:294] "Event occurred" object="events-2864/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288" kind="Pod" apiVersion="v1" type="Normal" reason="Created" message="Created container p"
Nov 17 10:23:07 kind-worker2 kubelet[250]: I1117 10:23:07.790106     250 event.go:294] "Event occurred" object="events-2864/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288" kind="Pod" apiVersion="v1" type="Normal" reason="Started" message="Started container p"
Nov 17 10:24:22 kind-worker2 kubelet[250]: I1117 10:24:22.021391     250 event.go:294] "Event occurred" object="events-2864/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288" kind="Pod" apiVersion="v1" type="Normal" reason="Killing" message="Stopping container p"
Nov 17 10:24:47 kind-worker2 kubelet[250]: E1117 10:24:47.315947     250 event.go:267] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"concurrent-27285743-8r8n2.16b84e5286d0bb4b", GenerateName:"", Namespace:"cronjob-6283", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Pod", Namespace:"cronjob-6283", Name:"concurrent-27285743-8r8n2", UID:"d9f4a985-eade-49e9-9e87-ec52d0f52510", APIVersion:"v1", ResourceVersion:"16831", FieldPath:""}, Reason:"FailedMount", Message:"MountVolume.SetUp failed for volume \"kube-api-access-2dgzb\" : failed to fetch token: serviceaccounts \"default\" is forbidden: unable to create new content in namespace cronjob-6283 because it is being terminated", Source:v1.EventSource{Component:"kubelet", Host:"kind-worker2"}, FirstTimestamp:time.Date(2021, time.November, 17, 10, 23, 0, 905286475, time.Local), LastTimestamp:time.Date(2021, time.November, 17, 10, 23, 0, 905286475, time.Local), Count:1, Type:"Warning", EventTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'events "concurrent-27285743-8r8n2.16b84e5286d0bb4b" is forbidden: unable to create new content in namespace cronjob-6283 because it is being terminated' (will not retry!)
```


there is only a few seconds delay in a working job
```
0.log.20211117-100007:2021-11-17T09:59:07.24986063Z stderr F I1117 09:59:07.249722       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-7897/events" latency="8.40999ms" userAgent="kube-scheduler/v1.23.0 (linux/amd64) kubernetes/2985b51/scheduler" audit-ID="7dbc1c7f-d3fb-46b2-bc5b-ab28f365feff" srcIP="172.18.0.2:37904" resp=201
0.log.20211117-100007:2021-11-17T09:59:14.919957639Z stderr F I1117 09:59:14.919827       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-7897/events" latency="2.354121ms" userAgent="kubelet/v1.23.0 (linux/amd64) kubernetes/2985b51" audit-ID="a9d1d8ed-08bf-4f49-9f9d-8e0fcb569981" srcIP="172.18.0.3:58886" resp=201
0.log.20211117-100007:2021-11-17T09:59:15.331432964Z stderr F I1117 09:59:15.331303       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-7897/events" latency="13.51088ms" userAgent="kubelet/v1.23.0 (linux/amd64) kubernetes/2985b51" audit-ID="b327d79d-1d45-4480-9a92-a036d0c4b8d5" srcIP="172.18.0.3:58886" resp=201
0.log.20211117-100007:2021-11-17T09:59:18.519263477Z stderr F I1117 09:59:18.519165       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-7897/events" latency="2.082386ms" userAgent="kubelet/v1.23.0 (linux/amd64) kubernetes/2985b51" audit-ID="f4c21594-088c-4b9d-a7a4-fbe2e522642b" srcIP="172.18.0.3:58886" resp=201
```

**Comment by BenTheElder** [comment]
I could've sworn we'd like banned conformance tests from depending on event delivery because it's not guaranteed to be reliable?

**Comment by aojea** [comment]
@liggitt right now the test waits 60 seconds for the event
https://github.com/kubernetes/kubernetes/blob/7f756417ff13f4d3a67862da4fa8f2a74f4887e3/test/e2e/node/events.go#L117-L118

since events are not synchronous, should we have a more reasonable period?

**Comment by spiffxp** [comment]
@BenTheElder I'm not familiar with this test, but you're right that Events and Reason aren't things that a Conformance test can depend upon

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md

>Conformance tests are intended to be stable and backwards compatible according to the standard API deprecation policies. Therefore any test that relies on specific output that is not subject to the deprecation policy cannot be promoted to conformance. Examples of tests which are not eligible to conformance:
>- anything that checks specific Events are generated, as we make no guarantees about the contents of events, nor their delivery
>   - If a test depends on events it is recommended to change the test to use an informer pattern and watch specific resource changes instead.
>   - An exception to this is tests that generates synthetic events themselves to verify that the API is capable of being exercised
>- anything that checks optional Condition fields, such as Reason or Message, as these may change over time (however it is reasonable to verify these fields exist or are non-empty)
>   - If the test is checking for specific conditions or reasons, it is considered overly specific and it is recommended to simply look for pass/failure criteria where possible, and output the condition/reason for debugging purposes only.

**Comment by spiffxp** [comment]
Did some digging.  It looks like this has been tagged as Conformance since at least 1.2 https://github.com/kubernetes/kubernetes/pull/15228 (sometime prior to 2015-10-07)

So this appears to be a case where we missed going back through the existing corpus of conformance tests when we wrote the above doc.  https://github.com/kubernetes/community/pull/2459 is where the rule originated (merged 2018-10-08)

**Comment by ehashman** [comment]
Code itself for corresponding test says 1.9: https://github.com/kubernetes/kubernetes/blob/7f756417ff13f4d3a67862da4fa8f2a74f4887e3/test/e2e/node/events.go#L39-L44

Git history shows the test basically hasn't changed in >1y: https://github.com/kubernetes/kubernetes/commits/master/test/e2e/node/events.go

**Comment by spiffxp** [comment]
I'll split out demotion into a separate PR, don't want to tangle this issue up

**Comment by ehashman** [comment]
The first single instance of a timeout failure I see is on https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460880476934770688

I'm going to see if I can try to bisect from there, probably merged recently.

**Comment by BenTheElder** [comment]
to be clear: the test has been around and we should look at fixing it, I was just surprised to see the test name with `[Conformance]`, we should open a different thread/issue on that topic 😅 

I am working on the kubernetes deb/RPM releases for the day or else I'd be taking a look.

**Comment by BenTheElder** [comment]
https://testgrid.k8s.io/conformance-kind#conformance,%20master%20(dev)%20[non-serial]&width=5
https://testgrid.k8s.io/conformance-kind#kind%20(IPv6),%20master%20(dev)%20%5Bnon-serial%5D&width=5

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/kubernetes/issues/106512 to cover the conformance demotion

**Comment by ehashman** [comment]
/priority critical-urgent

I am proceeding as though we have a perf regression with kubelet events until proven otherwise. Others feel free to also TAL.

**Comment by ehashman** [comment]
```
Ran 1 of 7031 Specs in 6.077 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 7030 Skipped
PASS

All tests passed...
Will keep running them until they fail.
This was attempt #57
No, seriously... you can probably stop now.
```

😒️ I can't reproduce locally on its own, I either will need to stress test it in parallel with some other stuff or I need a slower machine.

**Comment by ehashman** [comment]
/reopen

I don't want to just close this because the timeout was bumped if there's a possible regression

**Comment by ehashman** [comment]
Staring at @aojea's notes:

> the apiserver logs shows the kubelet events come too late, the test times out, is like they are queued in the kubelet
> 
> the test fails at 10:24:22 but the event arrives at 10:24:27
> 
> ```
> ï¿½[1mSTEPï¿½[0m: checking for scheduler event about the pod
> Nov 17 10:23:22.009: INFO: Saw scheduler event for our pod.
> ï¿½[1mSTEPï¿½[0m: checking for kubelet event about the pod
> Nov 17 10:24:22.015: FAIL: Unexpected error:
>     <*errors.errorString | 0xc00024a250>: {
> ```
> 
> https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460909846491566080
> 
> but kubelet times doesn't add up, there are vents generated by kubelet at 10:23:07 https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460909846491566080/artifacts/logs/kind-worker2/journal.log

Looking at the journal logs I'm seeing some (mild, not unexpected) client-side throttling:

```
Nov 17 10:23:17 kind-worker2 kubelet[250]: I1117 10:23:17.903811     250 request.go:597] Waited for 381.187219ms due to client-side throttling, not priority and fairness, request: GET:https://kind-control-plane:6443/api/v1/namespaces/events-2864/pods/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288
Nov 17 10:23:18 kind-worker2 kubelet[250]: I1117 10:23:18.102894     250 request.go:597] Waited for 195.73769ms due to client-side throttling, not priority and fairness, request: PATCH:https://kind-control-plane:6443/api/v1/namespaces/events-2864/pods/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288/status
Nov 17 10:23:18 kind-worker2 kubelet[250]: I1117 10:23:18.120423     250 status_manager.go:624] "Patch status for pod" pod="events-2864/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288" patch="{\"metadata\":{\"uid\":\"4516fd35-31f3-44ba-ada2-38beb016e168\"},\"status\":{\"$setElementOrder/conditions\":[{\"type\":\"Initialized\"},{\"type\":\"Ready\"},{\"type\":\"ContainersReady\"},{\"type\":\"PodScheduled\"}],\"conditions\":[{\"lastProbeTime\":null,\"lastTransitionTime\":\"2021-11-17T10:23:05Z\",\"status\":\"True\",\"type\":\"Initialized\"},{\"lastProbeTime\":null,\"lastTransitionTime\":\"2021-11-17T10:23:07Z\",\"status\":\"True\",\"type\":\"Ready\"},{\"lastProbeTime\":null,\"lastTransitionTime\":\"2021-11-17T10:23:07Z\",\"status\":\"True\",\"type\":\"ContainersReady\"}],\"containerStatuses\":[{\"containerID\":\"containerd://d1fe9a1fbc5c4a99fba85becec560f657158ad1ad7cba27b8b5eb7505414a7fc\",\"image\":\"k8s.gcr.io/e2e-test-images/agnhost:2.33\",\"imageID\":\"k8s.gcr.io/e2e-test-images/agnhost@sha256:5b3a9f1c71c09c00649d8374224642ff7029ce91a721ec9132e6ed45fa73fd43\",\"lastState\":{},\"name\":\"p\",\"ready\":true,\"restartCount\":0,\"started\":true,\"state\":{\"running\":{\"startedAt\":\"2021-11-17T10:23:07Z\"}}}],\"hostIP\":\"172.18.0.2\",\"phase\":\"Running\",\"podIP\":\"10.244.1.59\",\"podIPs\":[{\"ip\":\"10.244.1.59\"}],\"startTime\":\"2021-11-17T10:23:05Z\"}}"
Nov 17 10:23:18 kind-worker2 kubelet[250]: I1117 10:23:18.120570     250 status_manager.go:633] "Status for pod updated successfully" pod="events-2864/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288" statusVersion=2 status={Phase:Running Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-11-17 10:23:05 +0000 UTC Reason: Message:} {Type:Ready Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-11-17 10:23:07 +0000 UTC Reason: Message:} {Type:ContainersReady Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-11-17 10:23:07 +0000 UTC Reason: Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-11-17 10:23:05 +0000 UTC Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:172.18.0.2 PodIP:10.244.1.59 PodIPs:[{IP:10.244.1.59}] StartTime:2021-11-17 10:23:05 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:p State:{Waiting:nil Running:&ContainerStateRunning{StartedAt:2021-11-17 10:23:07 +0000 UTC,} Terminated:nil} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:true RestartCount:0 Image:k8s.gcr.io/e2e-test-images/agnhost:2.33 ImageID:k8s.gcr.io/e2e-test-images/agnhost@sha256:5b3a9f1c71c09c00649d8374224642ff7029ce91a721ec9132e6ed45fa73fd43 ContainerID:containerd://d1fe9a1fbc5c4a99fba85becec560f657158ad1ad7cba27b8b5eb7505414a7fc Started:0xc002656df9}] QOSClass:BestEffort EphemeralContainerStatuses:[]}
```

So that looks to me like the kubelet thinks it successfully got the event out by 10:23:18 which is well within our 1m timeout.

The timestamps in Antonio's post only look at POST, let me see if I can find the PATCH logs from apiserver

**Comment by ehashman** [comment]
From https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/106448/pull-kubernetes-conformance-kind-ga-only-parallel/1460909846491566080/artifacts/logs/kind-control-plane/pods/kube-system_kube-apiserver-kind-control-plane_d4f67f6be2fb429d5c53506800d50ddf/kube-apiserver/0.log.20211117-102458

```
2021-11-17T10:23:18.119584728Z stderr F I1117 10:23:18.119434       1 httplog.go:129] "HTTP" verb="PATCH" URI="/api/v1/namespaces/events-2864/pods/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288/status" latency="15.750065ms" userAgent="Go-http-client/2.0" audit-ID="b7a2a013-16bb-4ced-97ba-43d73c65f3eb" srcIP="172.18.0.2:58228" resp=200
```

That PATCH was received when the kubelet sent it so the event should have been updated and available... could there be an apiserver latency issue?

/sig api-machinery

**Comment by jackfrancis** [comment]
Possible repro on a just-rebased PR:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103069/pull-kubernetes-conformance-kind-ga-only-parallel/1461114630502879232

**Comment by ehashman** [comment]
> the test fails at 10:24:22 but the event arrives at 10:24:27

I see another PATCH at

```
2021-11-17T10:24:53.231326389Z stderr F I1117 10:24:53.231164       1 httplog.go:129] "HTTP" verb="PATCH" URI="/api/v1/namespaces/events-2864/pods/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288/status" latency="3.935179ms" userAgent="Go-http-client/2.0" audit-ID="9fc5aaa8-04af-456f-ad8b-c029714326f1" srcIP="172.18.0.2:58228" resp=200
```

but I think that's just from the pod getting torn down, it matches the Kubelet logs

**Comment by aojea** [comment]
> but I think that's just from the pod getting torn down, it matches the Kubelet logs
> URI="/api/v1/namespaces/events-2864/pods/send-events-aef99c99-2a7b-4c53-90ac-e0f0a7f85288/status" 

this is patching the pod status, probably setting the IPs or something like that, check the URI format is a patch on the pod objetct

> 2021-11-17T10:24:52.11568577Z stderr F I1117 10:24:52.115491       1 httplog.go:129] "HTTP" verb="POST" URI="/api/v1/namespaces/events-2864/events"

AFAIK those POST are creating the events , the URI format /api/v1/namespace/<namespace>/events ,

**Comment by aojea** [comment]
latest failure 1:45 am CET

https://storage.googleapis.com/k8s-triage/index.html?ci=0&pr=1&test=should%20be%20sent%20by%20kubelets%20and%20the%20scheduler

fix https://github.com/kubernetes/kubernetes/pull/106508 merged 1:55 AM CET, 


let's keep it open one day more

**Comment by ehashman** [comment]
I really don't think waiting from 1 -> 5m is an acceptable fix. This may be catching a performance regression.

It has stopped the bleeding but I think the problem remains...

**Comment by aojea** [comment]
> I really don't think waiting from 1 -> 5m is an acceptable fix. This may be catching a performance regression.

the events , by design, may take 6 mins , so it is completely legit that an event can take 5 mins, you cant' fail if take less (actually, by definition they are unreliable so it should be fine also to not receive it)

Performance can not be tested this way, there has to be an explicit test for that, and that test can not use events of course

**Comment by jackfrancis** [comment]
@aojea If the prior "median event lifecycle time" (in the context of this test) was previously < 1 min (and thus test flakes were rare) and now the "median event lifecycle time" has increased so as to trigger regular test flakes, that does seem like something interesting and worthy of prioritization.

It just seems probable that there's some "there" there.

This could also be explained by environmental changes, did anything change in the underlying test infra that correlates with the increased test failure rate?

**Comment by ehashman** [comment]
We previously had issues where presubmits were failing when periodics were not, see https://github.com/kubernetes/kubernetes/issues/97025

**Comment by ehashman** [comment]
> the events , by design, may take 6 mins , so it is completely legit that an event can take 5 mins, you cant' fail if take less (actually, by definition they are unreliable so it should be fine also to not receive it)
> 
> Performance can not be tested this way, there has to be an explicit test for that, and that test can not use events of course

If the test was newly introduced and was flaking, I'd agree with you, but we've had this test at 100% green since like the 1.2 release, and it only started flaking in the past 48h. This suggests to me that we merged a regression, but since it landed during code freeze, it's very difficult to determine which of the dozens of PRs it was.

**Comment by BenTheElder** [comment]
> This could also be explained by environmental changes, did anything change in the underlying test infra that correlates with the increased test failure rate?

Fairly confident there have not been substantial changes to the infra in the past week.
For blocking presubmits the infra is moved to public infra owned by the CNCF / SIG K8s infra and the configs are all in https://github.com/kubernetes/k8s.io

https://github.com/kubernetes/k8s.io/commits/main/infra/gcp shows the CI code being updated regularly, but not the build/test cluster that executes against.

KIND upgraded containerd/runc two weeks ago, but that doesn't fit. https://github.com/kubernetes-sigs/kind/commits/main

Given the timing it seems more like:
> [...] it only started flaking in the past 48h. This suggests to me that we merged a regression, but since it landed during code freeze, it's very difficult to determine which of the dozens of PRs it was.

**Comment by aojea** [comment]
> @aojea If the prior "median event lifecycle time" (in the context of this test) was previously < 1 min (and thus test flakes were rare) and now the "median event lifecycle time" has increased so as to trigger regular test flakes, that does seem like something interesting and worthy of prioritization.
> 
> It just seems probable that there's some "there" there.

> 
> This could also be explained by environmental changes, did anything change in the underlying test infra that correlates with the increased test failure rate?

indeed, check my analysis, https://github.com/kubernetes/kubernetes/issues/106506#issuecomment-972140283 a failing job has 429 FailedMount events, a job passing the test only 277, the rest of the events are the same 

Is not the first time that we have a performance regression or an env problem that causes similar effects, I remember one because of runc, one issue with the DNS on CI that elana linked, but this is not uncommon on the end of the release when CIs are busier

I don't disagree that there can be something, actually I totally agree, I disagree on using this test as a proof or measure of this "something", because:
1. this test verifies that, once you create a pod an event from the scheduler and from the kubelet is created.
2. events, looking at the code, can take 6 mins and by definition are not realiable ,

Tests should  be clear about what they test , for this cases we need to define a threshold, in sig network we have a test for latency of services, maybe adding one for latency of pods? we can define a threshold there, sig-scalability has it

**Comment by ehashman** [comment]
Latest in triage: https://storage.googleapis.com/k8s-triage/index.html?test=Events%20should%20be%20sent%20by%20kubelets%20and%20the%20scheduler%20about%20pods%20scheduling%20and%20running

It does appear that we had a bunch of CI flakes start showing up as well before increasing the timeouts in the test, so maybe not just a presubmit thing.

**Comment by aojea** [comment]
> 

Last failure 18/11 , no failure since the timeout bump merged, am I looking.in the wrong place?

**Comment by ehashman** [comment]
> Last failure 18/11 , no failure since the timeout bump merged, am I looking.in the wrong place?

No, it hasn't failed since the timeout bump merged, but the bump suppressing the symptom and hence is making this much more difficult to root cause. I'd suggest looking at some runs before the timeout was bumped?

**Comment by ehashman** [comment]
Per https://github.com/kubernetes/kubernetes/issues/106566#issuecomment-977217697 I'm wondering if some of the timing regressions we're seeing could be due to a perf regression having switched to CRI v1 with containerd. I can't find any corresponding failures on CRI-O.

/remove-priority critical-urgent
since test is no longer flaking

**Comment by ehashman** [comment]
/unassign
I'm not actively working this one

**Comment by ehashman** [comment]
I think this is similar to https://github.com/kubernetes/kubernetes/issues/106566#issuecomment-978229028 in that mainly dockershim jobs are affected.

**Comment by BenTheElder** [comment]
> 11/17/2021, 4:25:05 AM pr:pull-kubernetes-conformance-kind-ipv6-parallel
> 11/17/2021, 3:23:18 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel
> 11/17/2021, 1:58:59 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel
> 11/17/2021, 12:59:17 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel
> 11/17/2021, 12:02:17 AM pr:pull-kubernetes-conformance-kind-ga-only-parallel

FTR the jobs in the OP are not dockershim based, kubeadm's kinder jobs do often use dockershim but kind (not kinder) does not support dockershim.

https://storage.googleapis.com/k8s-triage/index.html?test=Events%20should%20be%20sent%20by%20kubelets%20and%20the%20scheduler%20about%20pods%20scheduling%20and%20running

is looking good though, fwiw

**Comment by aojea** [comment]
can we close this now?

**Comment by BenTheElder** [comment]
https://storage.googleapis.com/k8s-triage/index.html?test=Events%20should%20be%20sent%20by%20kubelets%20and%20the%20scheduler%20about%20pods%20scheduling%20and%20running

single 1.21 failure. i'm going to call this closed for now.
